src/datumaro/plugins/anchor_generator.py

```diff
@@ -0,0 +1,388 @@
+
+            return shifts
+
+        def get_anchors(
+            self, base_size: int, shifts: torch.Tensor, scales: torch.Tensor, ratios: torch.Tensor
+        ) -> torch.Tensor:
+            """This function is to create base anchors, which combinates ratios and scales.
+
+            Args:
+                base_size (int): Strides of feature map from feature pyramid network.
+                shifts (Tensor): Shift point coordinates in the original image space.
+                scales (Tensor): Scales for creating base anchors.
+                ratios (Tensor): Ratios for creating base anchors.
+
+            Returns:
+                Tensor: Set of anchor bounding box coordinates.
+            """
+            w, h = base_size, base_size
+            x_center, y_center = 0.5 * w, 0.5 * h
+
+            h_ratios = torch.sqrt(ratios)
+            w_ratios = 1 / h_ratios
+
+            ws = (w * w_ratios[:, None] * scales[None, :]).view(-1)
+            hs = (h * h_ratios[:, None] * scales[None, :]).view(-1)
+
+            base_anchors = [
+                x_center - 0.5 * ws,
+                y_center - 0.5 * hs,
+                x_center + 0.5 * ws,
+                y_center + 0.5 * hs,
+            ]
+            base_anchors = torch.stack(base_anchors, dim=-1)
+
+            anchors = base_anchors[None, :, :] + shifts[:, None, :]
+            anchors = anchors.view(-1, 4)
+
+            return anchors
+
+        def initialize(self, targets, scales, ratios):
+            init_ratios = []
+            init_scales = []
+            for level, base_size in enumerate(self.strides):
+                if len(scales[level]) == 1:
+                    representatives = [0.5]
+                elif len(scales[level]) == 2:
+                    representatives = [0.4, 0.6]
+                elif len(scales[level]) == 3:
+                    representatives = [0.25, 0.5, 0.75]
+                elif len(scales[level]) == 4:
+                    representatives = [0.25, 0.4, 0.6, 0.75]
+                elif len(scales[level]) == 5:
+                    representatives = [0.1, 0.25, 0.5, 0.75, 0.9]
+
+                anchors = self.get_anchors(
+                    base_size, self.shifts[level], scales[level], ratios[level]
+                )
+
+                overlaps = self.iou_calculator(targets, anchors)
+                gt_max_overlaps, _ = overlaps.max(dim=1)
+
+                scale_samples, ratio_samples = [], []
+                for target in targets[gt_max_overlaps > 0.1]:
+                    h = target[3] - target[1]
+                    w = target[2] - target[0]
+                    r = h / w
+                    affine_h = base_size * torch.sqrt(r)
+                    affine_w = base_size / torch.sqrt(r)
+                    s = max(h / affine_h, w / affine_w)
+
+                    scale_samples.append(s)
+                    ratio_samples.append(r)
```

Comment:
```suggestion
            scale_samples = [max(h / (base_size * torch.sqrt(r)), w / (base_size / torch.sqrt(r))) for h, w, r in zip(targets[:, 3] - targets[:, 1], targets[:, 2] - targets[:, 0], targets[:, 3] / targets[:, 2])]
            ratio_samples = [h / w for h, w in zip(targets[:, 3] - targets[:, 1], targets[:, 2] - targets[:, 0])]
```
I think there are some repeated computation.